### PR TITLE
chore: Add `restart` method to all token schemas

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes https://github.com/hashgraph/hedera-services/issues/13677
- Added `StakingInfoManagementSchema` to manage updating state with new nodes added during upgrade, by extracting from `V0490TokenSchema`
- This logic was added to `V0490TokenSchema`, but was missing in `V050TokenSchema`
- Updated  all token schemas to extend `StakingInfoManagementSchema` .

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/schemas/StakingInfoManagementSchema.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import com.hedera.node.config.data.StakingConfig;
+import com.swirlds.config.api.Configuration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Defines the schema for managing staking info.
+ * IMPORTANT: Every TokenSchema version should extend this interface
```
Comment (thomas-swirlds-labs) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/schemas/StakingInfoManagementSchema.java:39]: nit:  * IMPORTANT: Every TokenSchema version should extend this class

**Review by thomas-swirlds-labs — DISMISSED** [review]
LGTM @Neeharika-Sompalli !

**Review by tinker-michaelj — DISMISSED** [review]
LGTM, ty @Neeharika-Sompalli !

**Review by thomas-swirlds-labs — APPROVED** [review]

**Review by tinker-michaelj — DISMISSED** [review]
LGTM, ty @Neeharika-Sompalli !

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @Neeharika-Sompalli !

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/schemas/StakingInfoManagementSchema.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/schemas/StakingInfoManagementSchema.java:39]: ah yes Thomas , will fix it in next PR. Thank you!

**Pull request merged**